app/validators/unreserved_validator.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class UnreservedValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if value.nil?
```

Comment:
Could we put the call to this validator on a separate line in `Account` like:

```
validates :username, unreserved: true, if: :username?
```

so that we only bother calling this validator when a value is there?